drivers/backup/controllers/cluster/cluster_schedule.go

```diff
@@ -0,0 +1,148 @@
+package cluster
+
+import (
+	"fmt"
+	"github.com/portworx/torpedo/drivers/backup/utils"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/drivers/scheduler/k8s"
+	"github.com/portworx/torpedo/drivers/scheduler/spec"
+	"github.com/portworx/torpedo/pkg/log"
+	"github.com/portworx/torpedo/tests"
+	appsapi "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"reflect"
+)
+
+// ScheduleApplicationsConfig holds the necessary configuration to effectively schedule applications
+type ScheduleApplicationsConfig struct {
+	*scheduler.ScheduleOptions
+	instanceID string
+	controller *ClusterController
+}
+
+// validate validates ScheduleApplicationsConfig
+func (c *ScheduleApplicationsConfig) validate() error {
+	if len(c.Namespace) > GlobalMaxNamespaceCharLimit {
+		err := fmt.Errorf("namespace [%s] exceeds the maximum character limit of [%d]", c.Namespace, GlobalMaxNamespaceCharLimit)
+		return utils.ProcessError(err)
+	}
+	return nil
+}
+
+// getAppsSpec returns the slice of AppSpecs associated with the app-keys defined in the ScheduleApplicationsConfig
+func (c *ScheduleApplicationsConfig) getAppsSpec() ([]*spec.AppSpec, error) {
+	var appsSpec []*spec.AppSpec
+	if len(c.AppKeys) > 0 {
+		var specFactory *spec.Factory
+		switch driver := tests.Inst().S.(type) {
+		case *k8s.K8s:
+			specFactory = driver.SpecFactory
+		default:
+			var err error
+			specFactory, err = spec.NewFactory(tests.Inst().SpecDir, tests.Inst().V.String(), tests.Inst().S)
+			if err != nil {
+				debugMessage := fmt.Sprintf("spec-directory: [%s]; volume-driver: [%s]; scheduler: [%s]", tests.Inst().SpecDir, tests.Inst().V.String(), tests.Inst().S.String())
+				return nil, utils.ProcessError(err, debugMessage)
+			}
+		}
+		for _, appKey := range c.AppKeys {
+			appSpec, err := specFactory.Get(appKey)
+			if err != nil {
+				debugMessage := fmt.Sprintf("app-keys [%s]: app-key [%s]", c.AppKeys, appKey)
+				return nil, utils.ProcessError(err, debugMessage)
+			}
+			appsSpec = append(appsSpec, appSpec)
+		}
+	} else {
+		err := fmt.Errorf("the app list cannot be empty")
+		return nil, utils.ProcessError(err)
+	}
+	return appsSpec, nil
+}
+
+// getCustomAppsSpec returns a slice of customized AppsSpec by adding unique suffixes for spec names
+// getCustomAppsSpec currently supports only postgres and postgres-backup
+// TODO: Extend functionality to accommodate other applications
+func (c *ScheduleApplicationsConfig) getCustomAppsSpec() ([]*spec.AppSpec, error) {
```

Comment:
Make a list of apps which can use custom specs and for the rest go ahead with our usual scheduling without making changes to the app sepc.

Reply:
As discussed later we can add this for apps like cassandra

```
if obj, ok := spec.(*appsapi.StatefulSet); ok {
					obj.Spec.VolumeClaimTemplates[0].Annotations["volume.beta.kubernetes.io/storage-class"] += specNameSuffix
				}
```